lms/static/scripts/frontend_apps/utils/AuthWindow.js

```diff
@@ -43,11 +41,12 @@ export default class AuthWindow {
     const height = 560;
     const left = Math.round(window.screen.width / 2 - width / 2);
     const top = Math.round(window.screen.height / 2 - height / 2);
-    const settings = queryString
-      .stringify({ left, top, width, height })
-      .replace(/&/g, ',');
-    const authQuery = queryString.stringify({ authorization: this._authToken });
-    const authUrl = `${this._authUrl}?${authQuery}`;
+    const settings = `left=${left},top=${top},width=${width},height=${height}`;
+
+    const params = new URLSearchParams();
+    params.set('authorization', this._authToken);
+    const authUrl = `${this._authUrl}?${params}`;
```

Comment:
Pedant alert, but can this be `authURL` to match current casing conventions? (I realize this puts it out of step with `this._authUrl`)

Also, TIL, `URL.searchParams` is a _read-only_ `URLSearchParams` object. Was going to suggest using a `URL` object to munge this all together, but that is not useful, turns out. It's so very likely you already know this.

Reply:
> Pedant alert, but can this be authURL to match current casing conventions? (I realize this puts it out of step with this._authUrl)

Yes.

> Also, TIL, URL.searchParams is a read-only URLSearchParams object. Was going to suggest using a URL object to munge this all together, but that is not useful, turns out. It's so very likely you already know this.

Actually using a URL object is better here. It's more robust as it works even if `ltiLaunchURL` has query params already (though it currently does not). The MDN docs are confusing here because when they say `URL.searchParams` is read-only they mean that the `searchParams` _property_ cannot be re-assigned (ie. you can't do `url.searchParams = newParams`. The `URISearchParams` instance returned by `URL.searchParams` is not immutable though - you can modify the URL's query params using `url.searchParams.append(...)` etc.

Reply:
Oh, wow! Yeah, I was clicking around in those docs, which referred to the property as read-only but didn't give any further details. That makes sense! Thanks for the education.